pkg/sql/copy/copy_test: TestCopyFromRetries failed

pkg/sql/copy/copy_test.TestCopyFromRetries [failed](https://tanzanite.cluster.engflow.com/invocations/default/a9e5a759-1c55-4dfe-b5e1-8bdb9df15580?testReportRun=7&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NvcHk6Y29weV90ZXN0) on master @ [c24aedd67a89ea14e91b085d37103127d693aa1a](https://github.com/cockroachdb/cockroach/commits/c24aedd67a89ea14e91b085d37103127d693aa1a):


```
=== RUN   TestCopyFromRetries
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCopyFromRetries2687829098
    test_log_scope.go:81: use -show-logs to present logs inline
    copy_in_test.go:630: -- test log scope end --
test logs left over in: outputs.zip/logTestCopyFromRetries2687829098
--- FAIL: TestCopyFromRetries (101.87s)
=== RUN   TestCopyFromRetries/retries_successfully_on_every_batch_(errors_before)
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    copy_in_test.go:626: 
        	Error Trace:	pkg/sql/copy/copy_test/pkg/sql/copy/copy_in_test.go:626
        	Error:      	Received unexpected error:
        	            	sql: database is closed
        	Test:       	TestCopyFromRetries/retries_successfully_on_every_batch_(errors_before)
    --- FAIL: TestCopyFromRetries/retries_successfully_on_every_batch_(errors_before) (79.82s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCopyFromRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/copy/copy_in_test.go
+++ b/pkg/sql/copy/copy_in_test.go
@@ -35,7 +35,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/admission/admissionpb"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -429,7 +428,13 @@ func TestCopyFromRetries(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 117912)
+	// Ensure that the COPY batch size isn't too large (this test becomes too
+	// slow when metamorphic sql.CopyBatchRowSize is set to a relatively large
+	// number).
+	const maxCopyBatchRowSize = 1000
+	if sql.CopyBatchRowSize > maxCopyBatchRowSize {
+		sql.SetCopyFromBatchSize(maxCopyBatchRowSize)
+	}
 
 	// sql.CopyBatchRowSize can change depending on the metamorphic
 	// randomization, so we derive all rows counts from it.

